stream_buffer.c

```diff
@@ -360,7 +447,9 @@ static void prvInitialiseNewStreamBuffer( StreamBuffer_t * const pxStreamBuffer,
                                           pucStreamBufferStorageArea,
                                           xBufferSizeBytes,
                                           xTriggerLevelBytes,
-                                          ucFlags );
+                                          ucFlags,
```

Comment:
Should we maybe add a warning/assert if the `pxSendCompletedCallback` is non-NULL and `configUSE_SB_COMPLETED_CALLBACK == 0`?
This might help users debug why their callback is not being called and point them in the direction of the macro.

Reply:
The API allows you to create a stream buffer with user callback only if `configUSE_SB_COMPLETED_CALLBACK` is `1`

```
#if ( configUSE_SB_COMPLETED_CALLBACK == 1 )
    #define xStreamBufferCreateWithCallback( xBufferSizeBytes, xTriggerLevelBytes, pxSendCompletedCallback, pxReceiveCompletedCallback ) \
    xStreamBufferGenericCreate( xBufferSizeBytes, xTriggerLevelBytes, pdFALSE, pxSendCompletedCallback, pxReceiveCompletedCallback )
#endif
```